workflow/controller/controller.go

```diff
@@ -529,19 +529,20 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 					Value:     "true",
 				},
 			}
-			obj, _, _ := wfc.podInformer.GetIndexer().GetByKey(namespace + "/" + podName)
-			if p, ok := obj.(*apiv1.Pod); ok {
-				i := slices.Index(p.Finalizers, common.Finalizer)
-				if i >= 0 {
```

Comment:
- For some reason, the finalizer is already removed before this section.
- But the SharedInformer pod information is sometimes out of date and it looks like the pod still has the finalizer, so this condition is true even if the pod doesn't set the finalizer.

Reply:
So, it seems like getting an error back from the K8S API there isn't great (especially since the error is very generic and could be misleading) but at the same time I guess it's not harmful, right? Theoretically, the code should just move on (it looks like the caller isn't even checking the error), and the finalizer has already been removed. So, I'm not sure what happened after that as far as the `killContainers` logic...